flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/tableImpl.scala

```diff
@@ -133,61 +118,40 @@ class TableImpl(
       ExpressionParser.parseExpression(primaryKey))
   }
 
-  override def createTemporalTableFunction(
-      timeAttribute: Expression,
-      primaryKey: Expression)
-    : TemporalTableFunction = {
-    createTemporalTableFunctionInternal(
-      expressionBridge.bridge(timeAttribute),
-      expressionBridge.bridge(primaryKey))
+  def createTemporalTableFunction(
+    timeAttribute: Expression,
+    primaryKey: Expression)
+  : TemporalTableFunction = {
+    operationTreeBuilder.createTemporalTableFunction(
+      timeAttribute,
+      primaryKey,
+      logicalPlan)
   }
 
-  private def createTemporalTableFunctionInternal(
-      timeAttribute: PlannerExpression,
-      primaryKey: PlannerExpression)
-    : TemporalTableFunction = {
-    val temporalTable = TemporalTable(timeAttribute, primaryKey, logicalPlan)
-      .validate(tableEnv)
-      .asInstanceOf[TemporalTable]
-
-    TemporalTableFunctionImpl.create(
```

Comment:
Creating a temporal table function is unrelated to Operation, right? We should think about keeping the logic here, no?